packages/core/saved-objects/core-saved-objects-migration-server-internal/src/core/build_active_mappings.test.ts

```diff
@@ -208,3 +208,65 @@ describe('diffMappings', () => {
     expect(diffMappings(actual, expected)!.changedProp).toEqual('_meta');
   });
 });
+
+describe('getUpdatedHashes', () => {
+  test('is undefined if _meta is missing from actual', () => {
+    const actual: IndexMapping = {
+      dynamic: 'strict',
+      properties: {},
+    };
+    const expected: IndexMapping = {
+      _meta: {
+        migrationMappingPropertyHashes: { foo: 'bar' },
+      },
+      dynamic: 'strict',
+      properties: {},
+    };
+
+    expect(getUpdatedHashes(actual, expected)).toBeUndefined();
```

Comment:
I'm unsure about this. Probably an edge case but if an index doesn't have _meta can we safely conclude that all mappings have been picked up? Feels safer to assume all types' mappings have changed and need pickup?

Having `getUpdatedHashes(...): string[] | undefined` is also a slightly awkward API where `undefined` means something like "I don't know"...

Reply:
Actually, when this method returns `undefined` we don't pass in a `query`, so we already update all types.
But you've got a point, this can be confusing.
I'll update the logic to not rely on this `undefined`.